kong/plugins/request-termination/handler.lua

```diff
@@ -1,40 +1,82 @@
 local BasePlugin = require "kong.plugins.base_plugin"
+local singletons = require "kong.singletons"
 local responses = require "kong.tools.responses"
+local constants = require "kong.constants"
 local meta = require "kong.meta"
 
+
+local coroutine = coroutine
+local ngx = ngx
+
+
 local server_header = meta._NAME .. "/" .. meta._VERSION
 
+
 local RequestTerminationHandler = BasePlugin:extend()
 
+
 RequestTerminationHandler.PRIORITY = 2
-RequestTerminationHandler.VERSION = "0.1.0"
+RequestTerminationHandler.VERSION = "0.1.1"
+
+
+local function flush(ctx)
+  ctx = ctx or ngx.ctx
+
+  local response = ctx.delayed_response
+
+  local status       = response.status_code
+  local content      = response.content
+  local content_type = response.content_type
+  if not content_type then
+    content_type = "application/json; charset=utf-8";
+  end
+
+  ngx.status = status
+
+  if singletons.configuration.enabled_headers[constants.HEADERS.SERVER] then
+    ngx.header[constants.HEADERS.SERVER] = server_header
+
+  else
+    ngx.header[constants.HEADERS.SERVER] = nil
+  end
```

Comment:
This plugin should already have tests ensuring that this header is set/unset when terminating a request, and I am not seeing them. Perhaps this underlying implementation change warrants adding those tests.

Reply:
Added